src/lib/internal/helpers/builder.ts

```diff
@@ -103,228 +93,134 @@ export function effect<S extends Stores>(
-			index--;
-		});
-
-		// Function that attaches an event listener to an element
-		const attach: Attach = async (event, listener, options) => {
-			if (!isBrowser) return;
-			const element = await getElement();
-			if (!element) return;
-			unsubscribers.push(addEventListener(element, event, listener, options));
+export function lightable<T>(value: T): Readable<T> {
+	function subscribe(run: Subscriber<T>): Unsubscriber {
+		run(value);
+		return () => {
+			// don't need to unsub from anything
 		};
+	}
+	return { subscribe };
+}
 
-		// A function that returns the element associated with the current `id`
-		const getElement: Helpers['getElement'] = async () => {
-			if (!isBrowser) return null;
+type BuilderReturned<S extends Stores | undefined> = S extends Stores
+	? // eslint-disable-next-line @typescript-eslint/no-explicit-any
+	  (values: StoresValues<S>) => Record<string, any> | ((...args: any[]) => Record<string, any>)
+	: // eslint-disable-next-line @typescript-eslint/no-explicit-any
+	  () => Record<string, any> | ((...args: any[]) => Record<string, any>);
 
-			return await tick().then(() => getElementByMeltId(id));
-		};
+type BuilderArgs<
+	S extends Stores | undefined,
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	A extends Action<HTMLElement, any>,
+	R extends BuilderReturned<S>
+> = {
+	stores?: S;
+	action?: A;
+	returned?: R;
+};
 
-		const addAction: AddAction = async (action, parameters) => {
-			const element = await getElement();
-			if (!element) return;
+const isFunctionWithParams = (fn: unknown): fn is (...args: unknown[]) => Record<string, unknown> =>
+	typeof fn === 'function';
 
-			const ac = action(element, parameters);
-			if (ac) {
-				unsubscribers.push(function actionUnsub() {
-					ac.destroy?.();
+export function builder<
+	S extends Stores | undefined,
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	A extends Action<HTMLElement, any>,
+	R extends BuilderReturned<S>,
+	Name extends string
+>(name: Name, args?: BuilderArgs<S, A, R>) {
+	const { stores, action, returned } = args ?? {};
+
+	const derivedStore = (() => {
+		if (stores && returned) {
+			return derived(stores, (values) => {
+				const result = returned(values);
+				if (isFunctionWithParams(result)) {
+					const fn = (...args: Parameters<typeof result>) => {
+						return {
+							...result(...args),
+							[`data-melt-${name}`]: '',
+						};
+					};
+					fn.action = action ?? noop;
+					return fn;
+				}
+
+				return hiddenAction({
+					...result,
+					[`data-melt-${name}`]: '',
+					action: action ?? noop,
 				});
+			});
+		} else {
+			const returnedFn = returned as (() => R) | undefined;
+			const result = returnedFn?.();
+			if (isFunctionWithParams(result)) {
+				const resultFn = (...args: Parameters<typeof result>) => {
+					return {
+						...result(...args),
+						[`data-melt-${name}`]: '',
+					};
+				};
+				resultFn.action = action ?? noop;
+
+				return lightable(resultFn);
```

Comment:
But here we're just returning a `lightable`, which doesn't appear to filter out the action so it still shows the entire action string.

Reply:
It's not needed here, as the fn doesn't return an action. Same as on the derived above.